site/docs/tech/graph.md

```diff
@@ -0,0 +1,94 @@
+---
+id: graph
+title: Graph
+sidebar_label: Graph
+---
+
+#### General information
+Graph ([graph.go](https://github.com/perlin-network/wavelet/blob/master/graph.go)) is the module which holds [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) implementation used in the Wavelet.
+
+[Graph](https://github.com/perlin-network/wavelet/blob/master/graph.go#L74) is main structure which holds all DAG related information and implements all the functionality.
+All public/exported API is concurrently safe.
+
+`Graph` stores transactions in couple of different indices - in maps for quick access and some in balanced trees for quick search (e.g. parents and critical transactions).
+
+#### Adding transactions
+There are 3 reasons why transaction won't be added and action will result with error:
+- transaction already exists in the graph
+- depth of the transaction is below threshold (smaller than graph's root depth - [MaxDepthDiff](https://github.com/perlin-network/wavelet/blob/master/sys/const.go#L62)) - which basically means that transaction is too old for current graph's state
+- transaction is not valid (see [Transaction validation](#transactionvalidation))
+
+If none of the above is true, then transaction will be added to one of the indices, which won't affect graph structure, call it temporary transactions.
+In order for transaction to be part of the graph it should be "complete" - graph should have all of the parents of the transaction and they should be "complete" as well. This validation is performed only for transaction with depth below root depth by exactly [MaxParentsPerTransaction](https://github.com/perlin-network/wavelet/blob/master/sys/const.go#L65) rounds.
```

Comment:
`exactly x rounds` -> `exactly x levels`

Reply:
It doesn't look like a typo, more like a misused constant. @iwasaki-kenta ?
why levels instead of round though? i don't think we use term "level" anywhere in the system.

Reply:
It should be x levels/depths; rounds implies talking about consensus rounds rather than depths in the transaction graph. And also, you're correct - that is really strange: it should be `MaxDEpthDiff` instead of `MaxParentsPerTransaction`. Will get that fixed on the implementation side.